astropy/modeling/statistic.py

```diff
@@ -6,36 +6,163 @@
 # pylint: disable=invalid-name
 import numpy as np
 
-__all__ = ['leastsquare']
+from astropy.utils.decorators import format_doc
 
+__all__ = ["leastsquare", "leastsquare_1d", "leastsquare_2d", "leastsquare_3d"]
+
+
+def leastsquare(measured_vals, updated_model, weights, *x):
+    """
+    Least square statistic, with optional weights, in N-dimensions.
+
+    .. |ndarray| replace:: `~numpy.ndarray`
+
+    Parameters
+    ----------
+    measured_vals : |ndarray| or Sequence
+        Measured data values. Will be cast to array whose
+        shape must match the array-cast of the evaluated model.
+    updated_model : :class:`~astropy.modeling.Model` instance
+        Model with parameters set by the current iteration of the optimizer.
+        when evaluated on "x", must return array of shape "measured_vals"
+    weights : |ndarray| or None
+        Array of weights to apply to each residual.
+    *x : |ndarray|
+        Independent variables on which to evaluate the model.
+
+    Returns
+    -------
+    res : float
+        The sum of least squares.
+
+    Raises
+    ------
+    ValueError
+        If shape mismatch between model and measured
+    Exception
```

Comment:
This appears to refer to exceptions raised in the model call itself. It is no use if it isn't specific to exception type (and a vanilla exception would be unacceptable in this context). Remove it?

Reply:
✅ Sounds good. I had it in there because I had encountered it a few times and the traceback was long.